tests/contrib/mue/test_statearrangers.py

```diff
@@ -0,0 +1,141 @@
+import pytest
+import torch
+
+from pyro.contrib.mue.statearrangers import mg2k, profile
+
+
+def simpleprod(lst):
+    # Product of list of scalar tensors, as numpy would do it.
+    if len(lst) == 0:
+        return torch.tensor(1.)
+    else:
+        return torch.prod(torch.cat([elem[None] for elem in lst]))
+
+
+@pytest.mark.parametrize('M', [2, 20])
+@pytest.mark.parametrize('batch_size', [None, 5])
+@pytest.mark.parametrize('substitute', [False, True])
+def test_profile(M, batch_size, substitute):
+    torch.set_default_tensor_type('torch.DoubleTensor')
+
+    pf_arranger = profile(M)
+
+    u1 = torch.rand((M+1, 3))
+    u = torch.cat([(1-u1)[:, :, None], u1[:, :, None]], dim=2)
+    r1 = torch.rand((M+1, 3))
+    r = torch.cat([(1-r1)[:, :, None], r1[:, :, None]], dim=2)
+    s = torch.rand((M+1, 4))
+    s = s/torch.sum(s, dim=1, keepdim=True)
+    c = torch.rand((M+1, 4))
+    c = c/torch.sum(c, dim=1, keepdim=True)
+
+    if batch_size is not None:
+        s = torch.rand((batch_size, M+1, 4))
+        s = s/torch.sum(s, dim=2, keepdim=True)
+        u1 = torch.rand((batch_size, M+1, 3))
+        u = torch.cat([(1-u1)[:, :, :, None], u1[:, :, :, None]], dim=3)
+
+    if substitute:
+        ll = torch.rand((4, 5))
+        ll = ll/torch.sum(ll, dim=1, keepdim=True)
+        a0ln, aln, eln = pf_arranger.forward(torch.log(s), torch.log(c),
+                                             torch.log(r), torch.log(u),
+                                             torch.log(ll))
+    else:
+        a0ln, aln, eln = pf_arranger.forward(torch.log(s), torch.log(c),
+                                             torch.log(r), torch.log(u))
+
+    # - Remake transition matrices. -
```

Comment:
Could you add a little bit of commenting to this? Just like what each big code block is doing? And maybe a reference equation?

Also as we discussed, you might see if there is a simpler test for some special cases such as when indel prob goes to zero.